pkg/covermerger/covermerger.go

```diff
@@ -48,69 +46,94 @@ type Frame struct {
 	EndCol    int
 }
 
-func (fr FileRecord) Frame() Frame {
-	f := Frame{}
+func (fr FileRecord) Frame() (*Frame, error) {
+	f := &Frame{}
 	var err error
-	if f.StartCol, err = strconv.Atoi(fr[keyStartCol]); err != nil {
-		panic(fmt.Sprintf("failed to Atoi(%s)", fr[keyStartCol]))
+	if f.StartLine, err = strconv.Atoi(fr[KeyStartLine]); err != nil {
+		return nil, fmt.Errorf("failed to Atoi(%s): %w", fr[KeyStartLine], err)
 	}
-	if f.StartLine, err = strconv.Atoi(fr[keyStartLine]); err != nil {
-		panic(fmt.Sprintf("failed to Atoi(%s)", fr[keyStartLine]))
-	}
-	if f.EndCol, err = strconv.Atoi(fr[keyEndCol]); err != nil {
-		panic(fmt.Sprintf("failed to Atoi(%s)", fr[keyEndCol]))
-	}
-	if f.EndLine, err = strconv.Atoi(fr[keyEndLine]); err != nil {
-		panic(fmt.Sprintf("failed to Atoi(%s)", fr[keyEndLine]))
-	}
-	return f
+	return f, nil
 }
 
-func (fr FileRecord) HitCount() int {
-	if hitCount, err := strconv.Atoi(fr[keyHitCount]); err != nil {
-		panic(fmt.Sprintf("failed to Atoi(%s)", fr[keyHitCount]))
+func (fr FileRecord) HitCount() (int, error) {
+	if hitCount, err := strconv.Atoi(fr[KeyHitCount]); err != nil {
+		return 0, fmt.Errorf("failed to Atoi(%s): %w", fr[KeyHitCount], err)
 	} else {
-		return hitCount
+		return hitCount, nil
 	}
 }
 
 func (fr FileRecord) Arch() string {
-	return fr[keyArch]
+	return fr[KeyArch]
 }
 
 type MergeResult struct {
 	HitCounts  map[int]int
 	FileExists bool
+	LostFrames map[RepoBranchCommit]int64
 }
 
 type FileCoverageMerger interface {
-	AddRecord(rbc RepoBranchCommit, arch string, f Frame, hitCount int)
+	AddRecord(rbc RepoBranchCommit, arch string, f *Frame, hitCount int)
 	Result() *MergeResult
 }
 
-func batchFileData(c *Config, targetFilePath string, records FileRecords, base RepoBranchCommit,
+func batchFileData(c *Config, targetFilePath string, records FileRecords,
 ) (*MergeResult, error) {
+	log.Printf("processing %d records for %s", len(records), targetFilePath)
 	repoBranchCommitsMap := make(map[RepoBranchCommit]bool)
 	for _, record := range records {
 		repoBranchCommitsMap[record.RepoBranchCommit()] = true
 	}
-	repoBranchCommitsMap[base] = true
+	if c.BaseType == BaseManual {
+		repoBranchCommitsMap[c.Base] = true
+	}
 	repoBranchCommits := maps.Keys(repoBranchCommitsMap)
-	fileVersions, err := getFileVersions(c, targetFilePath, repoBranchCommits)
+	getFiles := getFileVersions
+	if c.getFileVersionsMock != nil {
+		getFiles = c.getFileVersionsMock
+	}
+	fvs, err := getFiles(c, targetFilePath, repoBranchCommits)
 	if err != nil {
 		return nil, fmt.Errorf("failed to getFileVersions: %w", err)
 	}
-	merger := makeFileLineCoverMerger(fileVersions, base)
+	base := getBaseRBC(c, targetFilePath, fvs)
+	merger := makeFileLineCoverMerger(fvs, base)
 	for _, record := range records {
+		var f *Frame
+		if f, err = record.Frame(); err != nil {
+			return nil, fmt.Errorf("error parsing %s records: %w", targetFilePath, err)
```

Comment:
You already prepend the filename in the caller of the `batchFileData` function.

Reply:
done